modules/sr/robot3/arduino.py

```diff
@@ -0,0 +1,289 @@
+"""The Arduino module provides an interface to the simulated Arduino."""
+from __future__ import annotations
+
+import logging
+from enum import Enum, IntEnum
+from types import MappingProxyType
+
+from controller import Robot
+from sr.robot3.utils import BoardIdentity
+from sr.robot3.arduino_devices import (
+    Led,
+    EmptyPin,
+    PinDevice,
+    Microswitch,
+    DistanceSensor,
+    PressureSensor,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class GPIOPinMode(str, Enum):
+    """The possible modes for a GPIO pin."""
+    INPUT = 'INPUT'
+    INPUT_PULLUP = 'INPUT_PULLUP'
+    OUTPUT = 'OUTPUT'
+
+
+class AnalogPins(IntEnum):
+    """The analog pins on the Arduino."""
+    A0 = 14
+    A1 = 15
+    A2 = 16
+    A3 = 17
+    A4 = 18
+    A5 = 19
+
+    # Note: these pins don't exist on a real arduino.
+    A6 = 20
+    A7 = 21
+
+
+DIGITAL_READ_MODES = {GPIOPinMode.INPUT, GPIOPinMode.INPUT_PULLUP, GPIOPinMode.OUTPUT}
+DIGITAL_WRITE_MODES = {GPIOPinMode.OUTPUT}
+ANALOG_READ_MODES = {GPIOPinMode.INPUT}
+
+
+def init_arduinos(webot: Robot) -> MappingProxyType[str, Arduino]:
+    serial_number = '1234567890'
+    return MappingProxyType({
+        serial_number: Arduino(webot, serial_number),
+    })
+
+
+class Arduino:
+    """
+    The Arduino board interface.
+
+    This is intended to be used with Arduino Uno boards running the sbot firmware.
+
+    :param webot: The robot object to connect to devices on.
+    :param serial_number: The serial number of the board.
+    """
+    __slots__ = ('_serial_num', '_serial', '_pins', '_identity')
+
+    @staticmethod
+    def get_board_type() -> str:
+        """
+        Return the type of the board.
+
+        :return: The literal string 'Arduino'.
+        """
+        return 'Arduino'
```

Comment:
This is redundant in the simulator. (It's also kinda wrong)
```suggestion
```

Reply:
As above, this function is present in the kit API.